apollo-federation/src/sources/connect/json_selection/README.md

```diff
@@ -560,6 +577,52 @@ character (so `.data { id name }`, or even `.data.id` or `.data.name`) to mean
 the same thing as `$.`, but this is no longer recommended, since `.data` is easy
 to mistype and misread, compared to `$.data`.
 
+### `AtPath ::=`
+
+![AtPath](./grammar/AtPath.svg)
+
+Similar to the special `$` variable, the `@` character always represents the
+current value being processed, which is often equal to `$`, but may differ from
+the `$` variable when `@` is used within the arguments of `->` methods.
+
+For example, when you want to compute the logical conjunction of several
+properties of the current object, you can keep using `$` with different property
+selections:
+
+```graphql
+all: $.first->and($.second)->and($.third)
+```
+
+If the `$` variable were rebound to the input value received by the `->and`
+method, this style of method chaining would not work, because the `$.second`
+expression would attempt to select a `second` property from the value of
+`$.first`. Instead, the `$` remains bound to the same value received by the
+closest enclosing `{...}` selection set, or the root value when used at the top
+level of a `JSONSelection`.
```

Comment:
I agree we need to explain the behavior/role of `$` earlier.

As for the reason/explanation, starting from the version of the language that includes the `$` variable, I've been hoping we could recommend `$.data` as the shorthand for a `PathSelection` consisting of a single `data` key, which is slightly different from selecting `data`, because that gives you back an object with a `data` property rather than the value of the `$.data` property. We still support `.data` (no `$`) for that purpose, since our older examples used that syntax, but I think it makes sense for all field/key-selecting `PathSelection`s to be shorthand for the same path beginning with `$.`, logically speaking.

So `.data` is the same as `$.data`, but also `nested.path` is the same as `.nested.path` is the same as `$.nested.path`.

Reply:
So there are no cases where you can use `$.data` or `.data` but not `data`? And the only time that `data` might be referring to something other than `$.data` is in an alias, right?

Reply:
Following GraphQL syntax, `data` by itself selects a field named "data" and includes the field name in the resulting object.

In other words, you can think of `data` is a shorthand for `data: $.data` (or `data: .data`), and `data { name }` would then be a shorthand for `data: $.data { name: $.name }` (note `$` gets rebound by the braces).